src/science/roughness/cable_roughness.F90

```diff
@@ -35,19 +34,100 @@ MODULE cable_roughness_module
+! and the aerodynamic contribution to the resistances controlling the fluxes 
+! of momentum, heat and water vapour between the land and atmosphere for each 
+! land point.
+!
+! The formulations take into account vegetation and snow cover. The dependence ! on atmospheric conditions (surface heat fluxes) is incorporated later within ! the [[cable_canopy_module]].
+! 
+!## Scientific description
+!
+! The scientific basis for the formulae is a combination of Localised Near 
+! Field theory for aerodynamic transfer within canopies (for the heat and 
+! water vapour fluxes) and bulk formulae for the roughness length and 
+! displacement height accounting for roughness sublayer effects (for the 
+! momentum flux).
+!
+! The calculations take into account whether:
+! 
+! 1. the soil model used is SLI or the default (soilsnow)
+! 2. the land point is vegetated (LAI > LAI_THRESH) or not
+!
+! The primary references for these formulations are
+!
+! * [Raupach MR (1989)](https://doi.org/10.1002/qj.49711548710)
+! * [Raupach MR (1992)](https://doi.org/10.1007/BF00155203)
+! * [Raupach MR (1994)](https://doi.org/10.1007/BF00709229)
+! * [Kowalczyk et al. (2006)](http://www.cmar.csiro.au/e-print/open/kowalczykea_2006a.pdf)
+!
+! <br></br>
+! The principal outputs from the MODULE are
+!
+! * `rough%zref_tq`: height above the displacement height for the 
+!    air temperature and humidity (m)
+! * `rough%zref_uv`: height above the displacement height for the 
+!    wind speeds (m)
+! * `rough%hruff`: the canopy height accounting for the presence 
+!    of snow (m)
+! * `canopy%vlaiw`: the leaf area index accounting for the presence
+!    of snow (\(m^2/m^2\))
+! * `rough%z0soil`: the aerodynamic roughness length for soil (m)
+! * `rough%z0ssoilsn`: the aerodynamic roughness lenght for snow (m)
+! * `roug%z0m`: the aerodynamic roughness length for the surface 
+!    (canopy+soil+snow) (m)
+! * `rough%disp`: the displacement height of the surface 
+!    (=0.0 if not vegetated) (m)
+! * `rough%zruffs`: the depth of the roughness sublayer for vegetated
+!    tiles (m)
+! * `rough%coexp`: the extinction coefficient for the wind speed profile 
+!    within the canopy, \(c_{0}\) (m). \(c_{0}\) is defined to be the 
+!    coefficient within an expontial wind profile, i.e.
+!    the wind speed at height \(z\) above the ground within a canopy of 
+!    height \(h_c\) is given by \( U(z) = U_{h} \exp\{ c_{0} (z-h_c) \} \) 
+!    where \(U_h\) is the wind speed at canopy top.
+! * `rough%usuh`: the ratio of the friction velocity, \(u_*\), to the wind 
+!    speed at canopy top (-)
+! * `rough%rt0us`: *normalised* aerodynamic resitance for the turbulent
+!    transfer from the soil/snow surface to the displacement height (-)
+! * `rough%rt1us`: *normalised* aerodynamic resistance for the turbulent 
+!    transfer from the displacement height to the reference level (-)
+!
+! The aerodynamic resistances for the current time step are evaluated later by dividing
```

Comment:
"Later": When, where? Too vague currently. Either add where this happens or scrap from the documentation here.

Reply:
Absolutely has to be included here - if we need more detail then change from 'later' to

'as rt0 and rough%rt1 in [[define_canopy]] by dividing ...'  

I actually want to refactor the whole code to make things more straightforward/simpler